internal/resource/obzone/obzone_task.go

```diff
@@ -396,3 +401,40 @@ func (m *OBZoneManager) MountBackupVolume() tasktypes.TaskError {
 	}
 	return nil
 }
+
+func (m *OBZoneManager) DeleteLegacyOBServer() tasktypes.TaskError {
+	operationManager, err := m.getOceanbaseOperationManager()
+	if err != nil {
+		return errors.Wrapf(err, "OBZone %s get oceanbase operation manager", m.OBZone.Name)
+	}
+	allOBServers, err := operationManager.ListServers()
+	if err != nil {
+		return errors.Wrap(err, "List observers in oceanbase")
+	}
+	observerList, err := m.listOBServers()
+	if err != nil {
+		return errors.Wrap(err, "List observer crs")
+	}
+	for _, observer := range allOBServers {
+		// skip observers in different obzone
+		if observer.Zone != m.OBZone.Spec.Topology.Zone {
+			continue
+		}
+		found := false
+		for _, observerCR := range observerList.Items {
+			if observer.Ip == observerCR.Status.PodIp {
```

Comment:
Maybe we should use `GetConnectAddr()` here

Reply:
Yes, after the privious PR for service mode merged, this should be changed to GetConnectAddr